Service/ZgwToVrijbrpService.php

```diff
@@ -243,13 +356,165 @@ private function getBirthProperties(array $zgw, array $output): array
+//                    break;
+//                case 'geslachtsnaam':
+//                    // Probably only for partner1.
+//                    $this->getCommitmentPartnerEigenschap($zaakEigenschappen, ['nameAfterCommitment', 'lastname'], $eigenschap);
+//                    break;
+                case 'bsn1':
+                    $this->getCommitmentWitness($zaakEigenschappen, 1, $eigenschappen);
+                    break;
+                case 'bsn2':
+                    $this->getCommitmentWitness($zaakEigenschappen, 2, $eigenschappen);
+                    break;
+                case 'bsn3':
+                    $this->getCommitmentWitness($zaakEigenschappen, 3, $eigenschappen);
+                    break;
+                case 'bsn4':
+                    $this->getCommitmentWitness($zaakEigenschappen, 4, $eigenschappen);
+                    break;
+                default:
+                    if (in_array($eigenschap->getValue('naam'), $properties) || in_array('all', $properties)) {
+                        $zaakEigenschappen[$eigenschap->getValue('naam')] = $eigenschap->getValue('waarde');
+                    }
+                    break;
+            }
+        }
+
+        return $zaakEigenschappen;
+    }//end getCommitmentZaakEigenschappen()
+
+    /**
+     * Adds a single Eigenschap value to the zaakEigenschappen array for Partner1 or Partner2 if it already is set for Partner1.
+     *
+     * @param array $zaakEigenschappen Array of key value pairs of the zaakEigenschappen of a Case.
+     * @param array $keys The keys we will check. Can be 1 or 2 strings used to check, used like this: [$key[0]][$key[1]].
+     * @param ObjectEntity $eigenschap An eigenschap ObjectEntity of a Case.
+     *
+     * @return void This function doesn't return anything.
+     */
+    private function getCommitmentPartnerEigenschap(array &$zaakEigenschappen, array $keys, ObjectEntity $eigenschap)
+    {
+        var_dump($keys);
+        if (count($keys) > 2 || count($keys) === 0) {
+            return;
+        }
+        if (count($keys) === 2) {
+            if (isset($zaakEigenschappen['partner1'][$keys[0]][$keys[1]]) === true) {
+                $zaakEigenschappen['partner2'][$keys[0]][$keys[1]] = $eigenschap->getValue('waarde');
+                return;
+            }
+            $zaakEigenschappen['partner1'][$keys[0]][$keys[1]] = $eigenschap->getValue('waarde');
+            return;
+        }
+
+        // If count($keys) == 1
+        // Todo: This with only 1 key is only used for bsn, so could be removed if we don't use it for bsn anymore
+        if (isset($zaakEigenschappen['partner1'][$keys[0]]) === true) {
+            $zaakEigenschappen['partner2'][$keys[0]] = $eigenschap->getValue('waarde');
+            return;
+        }
+        $zaakEigenschappen['partner1'][$keys[0]] = $eigenschap->getValue('waarde');
+    }//end getCommitmentPartnerEigenschap()
+
+    private function flattenEigenschappen(array $eigenschappen): array
```

Comment:
Remove this function and use getZaakEigenschappen() instead?

Reply:
getZaakEigenschappen resulted in 504's with witnesses